src/mixings/conditional_mixing.h

```diff
@@ -0,0 +1,32 @@
+#ifndef BAYESMIX_MIXINGS_CONDITIONAL_MIXING_H_
+#define BAYESMIX_MIXINGS_CONDITIONAL_MIXING_H_
+
+#include "base_mixing.h"
+
+class ConditionalMixing : public BaseMixing {

```

Comment:
I like the idea of having a `ConditionalMixing`.  I think that instead of calling `mass_xxx_cluster()`, conditional algorithms will simply call a `get_weights(covariate)`.